### DISCUSSION TITLE
Add a way for people to include prompts and details when uploading images

### DISCUSSION THREAD

**Discussion by JustMaier** [civitai/civitai#22] [Ideas]
A few services actually include this as metadata on the image upon generation. We could pre-populate by looking at that metadata and then allow people to adjust it.

The hard part about this is it would mean complicating the image upload experience which is very simple right now. Perhaps we could add a "Add Prompt" or "Add Details" button on the images next to the sort dragger, that would open a popover that allows them to set image properties like prompt etc...

Once we have prompt data, we could then expose it on images that have it with another "View Details" or "View Prompt" button that would open a popover with the details for the image. Perhaps in the lightbox we could even allow them to toggle a "Details" area on and off so that they can see the details as they browse. In the details popover, it'd be great to have a "Copy Prompt" button and maybe when we partner with image generators like [ArtBot](https://tinybots.net/artbot) we can have a button to "Generate Image".

- [x] add metadata when uploading images
- [x] ability to view image metadata on models and reviews
- [x] ability to view image metadata in lightbox

---

## It's live!

**Set metadata on images**
![image](https://user-images.githubusercontent.com/607609/202654724-c371830b-0a17-4035-b9ee-63392655f403.png)

**View metadata on images that have them**
![image](https://user-images.githubusercontent.com/607609/202654601-4f51b4ab-35a3-444a-8fec-7bd29a921169.png)

**View the metadata as you scroll through the lightbox**
![image](https://user-images.githubusercontent.com/607609/202654858-efc14fb5-c5f9-41a9-bcae-5c4511c9cccb.png)

**Metadata is automatically pulled in from images that have it!!!**
- Currently only Automatic1111 is supported.

**Comment by JustMaier** [comment]
## editing details on images:
![image](https://user-images.githubusercontent.com/607609/201738091-92de6778-1e9e-4098-a391-e8af5992c53b.png)

- The `i` Icon is gray unless there are values set
- Clicking on the `i` icon opens a pop-over with the image details fields
- On image upload, we attempt to pull the metadata out of the image (example of this incoming)

## Viewing details on image:
![image](https://user-images.githubusercontent.com/607609/201738149-dc71555f-b230-46b6-8ea5-4d31f9f493a4.png)

## Viewing details in lightbox:
![image](https://user-images.githubusercontent.com/607609/201738199-e0e7e8c1-69be-4144-afad-502003208db3.png)

- The `i` Icon on the lightbox would toggle on the details panel. Once toggled it would stay on and show the details of the images as you page through the lightbox

**Comment by bkdiehl** [comment]
@bkdiehl is working on this
